src/Raven.Server/Documents/Sharding/Subscriptions/ShardedSubscriptionBatch.cs

```diff
@@ -27,23 +34,59 @@ public void SetException(Exception e)
         ConfirmFromShardSubscriptionConnectionTcs.TrySetException(e);
     }
 
-    public ShardedSubscriptionBatch(RequestExecutor requestExecutor, string dbName, Logger logger) : base(requestExecutor, dbName, logger)
+    public ShardedSubscriptionBatch(RequestExecutor requestExecutor, string dbName, Logger logger, ShardedDatabaseContext databaseContext) : base(requestExecutor, dbName, logger)
     {
         ShardName = dbName;
+        _databaseContext = databaseContext;
     }
 
     protected override void EnsureDocumentId(BlittableJsonReaderObject item, string id) => throw new SubscriberErrorException($"Missing id property for {item}");
 
-    internal override void Initialize(BatchFromServer batch)
+    internal override async ValueTask InitializeAsync(BatchFromServer batch)
     {
         SendBatchToClientTcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
         ConfirmFromShardSubscriptionConnectionTcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
         
         ReturnContext = batch.ReturnContext;
+        Context = batch.Context;
         batch.ReturnContext = null; // move the release responsibility to the OrchestratedSubscriptionProcessor
-
-        base.Initialize(batch);
+        batch.Context = null;
```

Comment:
this feels wrong.
why do you need an additional context here?

Reply:
this is not an additional context, that context used  to write the missing includes on it